unstoppable-ios-app/domains-manager-ios/NetworkEnvironment/MobileAPIAccess.swift

```diff
@@ -523,12 +523,16 @@ extension NetworkService {
     }
     
     public func getActions(request: APIRequest) async throws -> NetworkService.ActionsResponse {
-        let data = try await fetchData(for: request.url,
-                                       body: request.body,
-                                       method: request.method,
-                                       extraHeaders: request.headers)
-        let response = try JSONDecoder().decode(ActionsResponse.self, from: data)
-        return response
+        do {
+            let data = try await fetchData(for: request.url,
+                                           body: request.body,
+                                           method: request.method,
+                                           extraHeaders: request.headers)
+            let response = try JSONDecoder().decode(ActionsResponse.self, from: data)
+            return response
+        } catch {
+            throw error
+        }
```

Comment:
What was the reason behind the decision of wrapping throwing function's body into do - catch block? How is this affects functionality?

Reply:
It's a trick for debugging -- so that I can catch the error still within this method